tests/post/steps/test_monitoring.py

```diff
@@ -119,10 +129,169 @@ def _check_object_exists():
     utils.retry(_check_object_exists, times=20, wait=3)
 
 
+@given(parsers.parse("the '{ns_name}' namespace exists"))
+def namespace_exists(k8s_client, ns_name):
+    client = k8s_client.resources.get(api_version="v1", kind="Namespace")
+    namespaces = client.get().items
+    try:
+        next(
+            filter(lambda ns: ns.get("metadata", {}).get("name") == ns_name, namespaces)
+        )
+    except StopIteration:
+        try:
+            client.create({"metadata": {"name": ns_name}})
+        except ApiException as err:
+            raise AssertionError(f"Unable to create namespace: {err}")
+
+
+@given(parsers.parse("the '{dash_name}' dashboard exists in folder '{folder_name}'"))
+def dashboard_exists(grafana_api, dash_name, folder_name):
+    folders = grafana_api.get_folders()
+    try:
+        folder = next(filter(lambda folder: folder["title"] == folder_name, folders))
+    except StopIteration:
+        raise AssertionError(f"no folder named '{folder_name}'")
+    dashboards = grafana_api.get_dashboards(folder_id=folder["id"])
+    try:
+        next(filter(lambda dash: dash["title"] == dash_name, dashboards))
+    except StopIteration:
+        raise AssertionError(
+            f"found no dashboard named '{dash_name}' in '{folder_name}'"
+        )
+    else:
+        return True
```

Comment:
Maybe to make it simpler we could simply use the uid directly since we can set this one manually in the dashboard.

To make this test really "realistic" we could generate an `uuid` at the beginning of the test (put it in the context) and then we create a dashboard using this `uuid` as uid